File: frontend/src/types/batch.ts
import type { ExportPreset } from './export'

export type UpscaleMethod = 'ffmpeg' | 'realesrgan'

export interface BatchExportRequest {
  media_ids: string[]
  preset: ExportPreset
}

export interface UpscaleRequest {
  media_id: string
  scale_factor: number
  method: UpscaleMethod
  output_name?: string
}

export interface BatchExportResponse {
  job_id: string
  child_job_ids: string[]
  status: string
  total: number
}

File: backend/app/schemas/batch.py
from typing import Literal

from pydantic import BaseModel

from app.schemas.export import ExportPreset

UpscaleMethod = Literal["ffmpeg", "realesrgan"]


class BatchExportRequest(BaseModel):
    media_ids: list[str]
    preset: ExportPreset


class UpscaleRequest(BaseModel):
    media_id: str
    scale_factor: int = 2  # 2 or 4
    method: UpscaleMethod = "ffmpeg"
    output_name: str = "upscaled.mp4"
